### PR TITLE
idempotent cancelling synchronization

### PR DESCRIPTION
**Pull request by informalict**
synchronize versions

### CODE REVIEW

File: pkg/replication/deployment_replication.go [hunk 1]
```diff
@@ -170,7 +172,7 @@ func (dr *DeploymentReplication) run() {
 			switch event.Type {
 			case eventArangoDeploymentReplicationUpdated:
 				if err := dr.handleArangoDeploymentReplicationUpdatedEvent(event); err != nil {
-					dr.reportDeploymentReplicationErr(err, "Failed to handle deployment replication update")
+					// It should not go into failed phase, maybe CR was not written to Kubernetes.
```
Comment (nikita-vanyasin) [inline pkg/replication/deployment_replication.go:175]: Maybe we can put at least some log.Warning() here?

File: pkg/replication/deployment_replication.go [hunk 2]
```diff
@@ -247,8 +249,14 @@ func (dr *DeploymentReplication) createEvent(evt *k8sutil.Event) {
 	dr.deps.EventRecorder.Event(evt.InvolvedObject, evt.Type, evt.Reason, evt.Message)
 }
 
-// Update the status of the API object from the internal status
+// Update the status of the API object from the internal status.
+// When object is being deleted then on internal status is affected.
```
Comment (nikita-vanyasin) [inline pkg/replication/deployment_replication.go:253]:
```suggestion
// Has no effect if object is being deleted.
```

File: pkg/replication/finalizers.go [hunk 3]
```diff
@@ -138,31 +143,75 @@ func (dr *DeploymentReplication) inspectFinalizerDeplReplStopSync(ctx context.Co
...
+			return false, errors.WithMessage(err, "Failed to create destination synchronization master client")
+		}
+
+		if util.BoolOrDefault(p.Spec.Cancellation.EnsureInSync, true) {
+			if inSync, inSyncShards, totalShards, err := dr.ensureInSync(ctx, destClient); err != nil {
+				return false, err
+			} else if !inSync {
+				if time.Since(dr.lastLog) > time.Second*5 {
+					dr.lastLog = time.Now()
+					dr.log.Info("Consistency is being checked, %d shards in sync out of %d", inSyncShards, totalShards)
```
Comment (nikita-vanyasin) [inline pkg/replication/finalizers.go:160]:
```suggestion
					dr.log.Info("Consistency is being checked, %d of %d shards are in-sync", inSyncShards, totalShards)
```

File: pkg/apis/replication/v1/conditions.go [hunk 4]
```diff
@@ -31,6 +31,10 @@ type ConditionType string
 const (
 	// ConditionTypeConfigured indicates that the replication has been configured.
 	ConditionTypeConfigured ConditionType = "Configured"
+	// ConditionTypeEnsuredInSync indicates that the replication consistency was checked.
+	ConditionTypeEnsuredInSync ConditionType = "EnsuredInSync"
+	// ConditionTypeAborted indicates that the replication is cancelling with abort option.
```
Comment (nikita-vanyasin) [inline pkg/apis/replication/v1/conditions.go:36]:
```suggestion
	// ConditionTypeAborted indicates that the replication was canceled with abort option.
```

**Review by nikita-vanyasin — COMMENTED** [review]
LGTM except breaking change. See other suggestions too.

File: pkg/replication/deployment_replication.go (hunk 1 above)
Comment (informalict) [reply pkg/replication/deployment_replication.go:175]: added

File: pkg/apis/replication/v1/replication_status.go [hunk 5]
```diff
@@ -36,10 +36,6 @@ type DeploymentReplicationStatus struct {
 	// Destination contains the detailed status of the destination endpoint
 	Destination EndpointStatus `json:"destination"`
 
-	// CancelFailures records the number of times that the configuration was canceled
```
[the remark this answers is not in the document]
Comment (informalict) [reply pkg/apis/replication/v1/replication_status.go:39]: marked as deprecated

**Comment by informalict** [comment]
There is new PR https://github.com/arangodb/kube-arangodb/pull/1134

**Pull request closed without merging**